compiler/rustc_middle/src/ty/erase_regions.rs

```diff
@@ -1,21 +1,27 @@
 use tracing::debug;
 
-use crate::query::Providers;
+use crate::dep_graph::DepGraph;
 use crate::ty::{
     self, Ty, TyCtxt, TypeFlags, TypeFoldable, TypeFolder, TypeSuperFoldable, TypeVisitableExt,
 };
 
-pub(super) fn provide(providers: &mut Providers) {
-    *providers = Providers { erase_regions_ty, ..*providers };
-}
+impl<'tcx> TyCtxt<'tcx> {
+    /// Erases regions from `ty` to yield a new type.
+    pub fn erase_regions_ty(self, ty: Ty<'tcx>) -> Ty<'tcx> {
```

Comment:
Could you add a comment telling that this function is expected to be a pure function of its input, and that's why it cannot call any other queries.